### PR TITLE
P-145 Add {para,side}chain runtime versions to VC

### PR DESCRIPTION
**Pull request by arieljbirnbaum**
### Context

Implementing P-145
Following VC schema changes added in [vc-jsonschema#24](https://github.com/litentry/vc-jsonschema/pull/24)

### How (Optional)

- Sidechain version is taken from the build time constant.
- Parachain version is taken from the `node_metadata_repo` argument at VC construction time.

### CODE REVIEW

**Review by Traf333 — COMMENTED** [review]
@grumpygreenguy should we exclude bitacross-worker since there are no sidechain and VC?

**Comment by Kailai-Wang** [comment]
> @grumpygreenguy should we exclude bitacross-worker since there are no sidechain and VC?

IMO yes - there should be no stf/sidechain for bc-worker (there's a PR to remove it)

File: tee-worker/core-primitives/stf-interface/src/lib.rs [hunk 1]
```diff
@@ -20,6 +20,12 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
```
Comment (silva-fj) [inline tee-worker/core-primitives/stf-interface/src/lib.rs]: Is this really needed?

File: tee-worker/core-primitives/node-api/metadata/src/lib.rs [hunk 2]
```diff
@@ -56,20 +56,21 @@ pub trait NodeMetadataTrait:
...
-			+ VCMPCallIndexes
-			+ SystemSs58Prefix
-			+ UtilityCallIndexes
-			+ ProxyCallIndexes
-			+ BalancesCallIndexes
-			+ TimestampCallIndexes,
-	> NodeMetadataTrait for T
-{
-}
+// impl<
```
Comment (silva-fj) [inline tee-worker/core-primitives/node-api/metadata/src/lib.rs]: Can this be removed?

**Review by silva-fj — COMMENTED** [review]
it looks like you had a lot of fun adding just two values 😆

File: tee-worker/core-primitives/node-api/metadata/src/lib.rs (hunk 2 above)
Comment (arieljbirnbaum) [reply tee-worker/core-primitives/node-api/metadata/src/lib.rs]: Possibly. @Kailai-Wang suggested having the method in the `SystemSs58Prefix` trait tho (which should then probably be renamed to sth like `SystemPalletTrait`) so I'm having a look at that; if that works then I can keep the old impl

File: tee-worker/core-primitives/stf-interface/src/lib.rs (hunk 1 above)
Comment (arieljbirnbaum) [reply tee-worker/core-primitives/stf-interface/src/lib.rs]: It _was_ when I was passing the version numbers as `String` (as opposed to `u32`) :D
Will try yeeting it; thanks for the catch :raised_hands:

File: tee-worker/core-primitives/node-api/metadata/src/lib.rs (hunk 2 above)
Comment (arieljbirnbaum) [reply tee-worker/core-primitives/node-api/metadata/src/lib.rs]: Fixed :D

File: tee-worker/core-primitives/stf-interface/src/lib.rs (hunk 1 above)
Comment (arieljbirnbaum) [reply tee-worker/core-primitives/stf-interface/src/lib.rs]: Yeeted :D

**Review by silva-fj — APPROVED** [review]
LGTM!

**Pull request merged**